connectors/sources/servicenow.py

```diff
@@ -671,28 +672,38 @@ async def _fetch_attachment_metadata(self, batched_apis, table_access_control):
             self._logger.warning(
                 f"Skipping batch data for {batched_apis}. Exception: {exception}."
             )
-
-        await self.queue.put(EndSignal.ATTACHMENT)
+        finally:
+            await self.queue.put(EndSignal.ATTACHMENT)
 
     async def _attachment_metadata_producer(self, record_ids, table_access_control):
-        attachment_apis = self.servicenow_client.get_attachment_apis(
-            url=ENDPOINTS["ATTACHMENT"], ids=record_ids
-        )
+        attachment_apis = None
+        try:
+            attachment_apis = self.servicenow_client.get_attachment_apis(
+                url=ENDPOINTS["ATTACHMENT"], ids=record_ids
+            )
 
-        for batched_apis_index in range(0, len(attachment_apis), ATTACHMENT_BATCH_SIZE):
-            batched_apis = attachment_apis[
-                batched_apis_index : (  # noqa
-                    batched_apis_index + ATTACHMENT_BATCH_SIZE
-                )
-            ]
-            await self.fetchers.put(
-                partial(
-                    self._fetch_attachment_metadata, batched_apis, table_access_control
+            for batched_apis_index in range(
+                0, len(attachment_apis), ATTACHMENT_BATCH_SIZE
+            ):
+                batched_apis = attachment_apis[
+                    batched_apis_index : (  # noqa
+                        batched_apis_index + ATTACHMENT_BATCH_SIZE
+                    )
+                ]
+                await self.fetchers.put(
+                    partial(
+                        self._fetch_attachment_metadata,
+                        batched_apis,
+                        table_access_control,
+                    )
                 )
+                self.task_count += 1
+        except Exception as exception:
```

Comment:
This feels like it should still fail the sync, no?

Reply:
Hey, I didn't understand what you are trying to convey. 

These changes are similar to the ones we implemented in confluence https://github.com/elastic/connectors/pull/2380. 

Using these changes we make sure that END_SIGNALS for a given task are added to queue even if error occurs inside that task, this prevents _consumer method from running in a forever while loop.

Reply:
In that linked PR, the caught exceptions are logged with `_logger.exception` and then re-`raise`'d. Here, you're just swallowing the error, so the sync won't fail after this is merged, but the same issue _would_ fail the sync today.

Unless you've got a good reason (which you might!), I suggest re-raising the error, and logging these as error/exception instead of warn.

Reply:
Got it, i'll update the code to re-raise the error

Reply:
@seanstory updated the PR with the changes.